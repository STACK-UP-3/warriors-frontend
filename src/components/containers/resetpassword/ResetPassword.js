import React, { Component } from "react";

class ResetPassword extends Component {
  constructor(props) {
    super(props);
    this.state = {
      password: "",
      confirmPassword: "",
    };
  }
  handlePasswordChange = (e) => {
    console.log(e.target.value);
    const { value: password } = e.target;
    this.setState({ password });
  }

  render() {
    return (
      <div className="main-reset">
        <div className="resetpasswordcontainer">
          <div className="inner-reset-container">
            <div className="txt-container">
              <p className="p-reset-message">Reset password</p>
              <input
                id="password"
                type="password"
                placeholder="Enter password"
                className="txt-input"
                value={this.state.password}
                onChange={this.handlePasswordChange}
              />
              <input
                id="confirmpassword"
                type="password"
                placeholder="Confirm password"
                className="txt-input"
                value={this.state.password}
                onChange={this.handlePasswordChange}
              />
              <div className="div-contaier">
                <button className="btn btn-cancel">Cancel</button>
                <button className="btn btn-send">Reset password</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default ResetPassword;
