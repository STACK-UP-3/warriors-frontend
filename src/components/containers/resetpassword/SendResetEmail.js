import React from "react";
import { sendResetEmail} from "../../../redux/actions/resetPasswordAction";
import { connect } from "react-redux";


class SendResetEmail extends React.Component {

 state={
   email:'',
   isResetPasswordSent:false
 }
 componentWillReceiveProps({isResetPasswordSent}){
    this.setState({
      isResetPasswordSent
    })
  }

  handleEmailChange=(e)=>{
   this.setState(
     {
       email:e.target.value
     }
   )
  }
  render() {
    const {email}=this.state;
    return (
      <div className="main-reset">
        <div className="resetpasswordcontainer">
          <div className="inner-reset-container">
            <div className="txt-container">
              <p className="p-reset-message">Forgot password</p>
              <input
                id="email"
                type="email"
                placeholder="Enter your email"
                className="txt-input"
                onChange={this.handleEmailChange}
                value={this.state.email}
              />
              <div className="div-contaier">
                <button className="btn btn-cancel">Cancel</button>
                <button className="btn btn-send" onClick={()=>{
                 this.props.sendResetEmail(email)
                }} >Send</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }

}
const mapStateToProps = ({ resetPassword: {isResetLinkSent} }) => ({
  isResetPasswordSent:isResetLinkSent
});

export default connect(mapStateToProps, { sendResetEmail})(SendResetEmail);
