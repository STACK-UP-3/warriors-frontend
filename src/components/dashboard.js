import React from 'react';

export default ()=>{

const token = localStorage.getItem('token');

return(
    <div>
        <h1> Welcome To Your Dashboard </h1>
        <p> Your token is: { token } </p>
    </div>
)};
