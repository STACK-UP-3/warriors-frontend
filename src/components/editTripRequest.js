import React from 'react';

export default (props)=>{
return(
<div>
      <h5>Lets EDIT TRIP REQUEST : { props.match.params.id }</h5>
</div>
)};
