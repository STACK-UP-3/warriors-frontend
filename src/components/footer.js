import React from 'react';

export default ()=>(
    
  <footer className="page-footer blue darken-3">
    <div className="container">
    <div className='footer'>
        <span>
          Â© 2020 BareFoot Nomad, All rights reserved.
        </span>
        </div>
    </div>
</footer>
);
