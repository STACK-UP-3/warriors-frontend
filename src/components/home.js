import React from 'react';
import M from 'materialize-css'; // Materialize JS ~ https://www.youtube.com/watch?v=EboLM8OjlP4

export default class Home extends React.Component {

  componentDidMount() {
    M.AutoInit();
  };

  render() {
    return (
      <div>
        <div className="row">
          <h1>BareFoot Nomad Welcomes You!!</h1>
          <a className="waves-effect waves-light btn" href="/login">Login</a>
        </div>
      </div>
    );
  };
}
