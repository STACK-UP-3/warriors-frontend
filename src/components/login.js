import React,{ useEffect } from 'react';
import { useSelector,useDispatch } from 'react-redux';
import M from 'materialize-css';
import { toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { onUserLogin,TokenAuth } from '../redux/actions/loginActions';
import LoginForm from './loginForm';
import Footer from './footer';
import Message from './requestFail';
import SideContainer from './sideContainer';
import SocialAuth from './socialAuth';

toast.configure();

export default ()=>{

    const loginStore = useSelector( store=> store.loginData );
    const dispatch = useDispatch();

    useEffect( ()=>{
        M.AutoInit();
        if(loginStore.error){
            toast.error( <Message />,{
                position: toast.POSITION.TOP_CENTER,
                toastId: 'loginFetchError',
            });
        }
        
        if(loginStore.userInfo.status === 200){
            const token = JSON.stringify(loginStore.userInfo.data.access_token);
            localStorage.setItem('token', token);
            window.location.assign('/dashboard');
        } 

        let token;
        const url  = window.location.search;
        const arr = url.split(""); 
        arr.splice(0,7); 
        
        if(arr.length){
            token = arr.join("")
        }

        if(localStorage.getItem("token")){
            token = localStorage.getItem("token");
        }
        

        if(token) return tokenAuthenticator(token)
    },[]);
    
    const tokenAuthenticator = (token) => { dispatch(TokenAuth(token)); }
    const onFormSubmit = data => { dispatch(onUserLogin(data));  }

    const loginsideContainerTitle = 'Welcome Back to BareFoot Nomad';
    const loginsideContainerSubtitle  = (
            <div>
                <p>
                     If you have not created an account please sign up to use our services
                </p>
                    <br />

                <h4
                    onClick={ ()=> window.location.assign('/signup') }
                >
                    sign up
                </h4>
            </div>
    );
    
    return (
        <div className='loginContainer'>
        <div className='main'>
            <SideContainer title={ loginsideContainerTitle } subtitle ={ loginsideContainerSubtitle  }/>
                
                <div className='formContainer'>

                    <h1> Sign in to BareFoot Nomad </h1>

                        <LoginForm 
                            onSubmit={ onFormSubmit } 
                            userInfo = { loginStore.userInfo } 
                            isLoading = {loginStore.loading}
                            testdata='loginForm' 
                        />
                            <br/>
                        <SocialAuth />
                    
                </div>
        </div>

        <Footer />


    </div>
    );
}
