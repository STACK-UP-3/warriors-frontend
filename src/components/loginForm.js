import React, { useState, useEffect } from 'react';
import SyncLoader from "react-spinners/SyncLoader";

export default (props)=>{

    const [formState, setFormSate] = useState({
        email:'',
        password:'',
        error: '',
    });

    useEffect(()=>{
        setFormSate({ ...formState, error: props.userInfo.message });
    }, [props.userInfo]);

    const onInputChange = (e)=>{
        const { id, value } = e.target;
        setFormSate({ ...formState, [id]:value, error: ''}); 
    }

    const onFormSubmit = (e) =>{
        e.preventDefault();
        const email = formState.email;
        const password = formState.password;
        
        if(email.match(/\S+@\S+\.\S+/)){
            props.onSubmit({ email, password });
        }else{
            setFormSate({ ...formState, error:'incorrect email format'});
        }
    }
    
    const forgotPassword = ()=> window.location.assign('/reset');

    return(
        <div>
            
            <form 
                onSubmit={ onFormSubmit } 
                className='col s12' 
                testdata="form"
            >
                    <div className='form-content'>

                        <div className='row' id='textInput'>
                            <div className='input-field col s12'>
                                <input 
                                    name = 'Email'
                                    id='email' 
                                    type='email' 
                                    className='validate'
                                    onChange={ onInputChange } 
                                    testdata="inputChange"
                                    disabled= { props.isLoading }
                                />
                                <label htmlFor='email'>Email</label>
                            </div>
                        </div>

                        <div className='row' id='textInput'>
                            <div className='input-field col s12'>
                                <input 
                                    name='Password'
                                    id='password' 
                                    type='password' 
                                    className='validate'
                                    onChange={ onInputChange }
                                    disabled= { props.isLoading }
                                />
                                <label htmlFor='password'>Password</label>
                            </div>
                        </div>

                        <div className='contentFooter'>
                        <span 
                            id='forgotPassword' 
                            onClick={ forgotPassword }
                        > 
                            Forgot password?
                        </span> 
                            <br/> <br/>
                        <span className='error'>
                            { formState.error ==='You have signed in successfully' ? null : (formState.error) } 
                        </span> 
                        </div>

                    </div> <br/>

                    <div className='form-footer'>
                        <div className="submitBtn">
                            <button 
                                className='btn waves-effect waves-light'
                                id='submit'
                                disabled={ !formState.email || !formState.password }
                            >
                                {props.isLoading?( 
                                <SyncLoader
                                    size={5}
                                    margin={5}
                                    color={"#fff"}
                                    loading={ props.isLoading }
                                /> 
                                ):'Login'}
                                
                    </button>
                </div>
            </div>
    </form>
</div>
    );
};
