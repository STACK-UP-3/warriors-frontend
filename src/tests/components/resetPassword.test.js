import React from "react";
import { shallow } from "enzyme";
import { resetpassword as PasswordReset, mapStateToProps } from "../../components/resetpassword";
let resetPasswordComp;
const token =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImRhbm55a2Ftb3NvMjAyM0BnbWFpbC5jb20iLCJpYXQiOjE1OTE2NTcxNzEsImV4cCI6MTU5MTY2MDc3MX0.aUwB5Q6TF6KjM8y6viuhQlnAbW_z3oji2v9g689dp1c";

describe("***************** Testing t resetpasword  component ***************", () => {
  const props = {
    isPasswordReset: true,
    resetPassword: jest.fn(),
    location: { search: { token: token } },
  };
  beforeAll(() => {
    resetPasswordComp = shallow(<PasswordReset {...props} />);
  });

  it("Should render the reset password page correctly", () => {
    resetPasswordComp
      .instance()
      .componentWillReceiveProps({ isPasswordReset: true });
    resetPasswordComp
      .instance()
      .handlePasswordChange({ target: { value: "passwosjjsj" } });
    resetPasswordComp.instance().handleResetPasswordEvent();
    expect(resetPasswordComp).toMatchSnapshot();
  });
  it("Should send states to props in reset password", () => {
    expect(
      typeof mapStateToProps({ resetPassword: { isPasswordReset: true } })
    ).toBe("object");
  });
});
