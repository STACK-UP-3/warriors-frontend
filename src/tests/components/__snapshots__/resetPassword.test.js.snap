// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`************* Testing t resetpasword  component *************** Should render the reset password page correctly 1`] = `
<div
  className="main-reset"
>
  <div
    className="resetpasswordcontainer"
  >
    <div
      className="inner-reset-container"
    >
      <div
        className="txt-container"
      >
        <p
          className="p-reset-message"
        >
          Reset password
        </p>
        <input
          className="txt-input"
          id="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
          value="passwosjjsj"
        />
        <input
          className="txt-input"
          id="confirmpassword"
          onChange={[Function]}
          placeholder="Confirm password"
          type="password"
          value="passwosjjsj"
        />
        <div
          className="div-contaier"
        >
          <Link
            className="btn btn-cancel"
            to="/login"
          >
            Cancel
          </Link>
          <button
            className="btn waves-effect waves-light btn-send"
            onClick={[Function]}
          >
            Reset password
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
